src/devices/bosch.ts

```diff
@@ -908,14 +917,90 @@ const definitions: Definition[] = [
+            convertSet: async (entity, key, value:smokeSensorData, meta) => {
+                const dataToSend = {
+                    cluster: value?.cluster,
+                    command: value?.command,
+                    payload: value?.payload,
+                };
+                if (key === 'smoke_alarm_state' || key==='intruder_alarm_state') {
+                    dataToSend.cluster = 'ssIasZone';
+                    dataToSend.payload = {data: value};
+                    dataToSend.command = 128;
+                }
+
+                //
+                // the data of the payload can either be decimal:
+                // 1: this disables the intruder alarm state
+                // 46081: this enables the intruder alert siren
+                // 256: this enables the smoke test alarm siren
+                const endpoint = meta.device.getEndpoint(1);
+                await endpoint.command(
+                    dataToSend.cluster,
+                    dataToSend.command,
+                    dataToSend.payload,
+                    {
+                        ...manufacturerOptions,
+                        ...utils.getOptions(meta.mapped, entity),
+                    },
+                );
+                return {state: {alarm_state: dataToSend.payload.data}};
+            },
+            convertGet: async (entity, key, meta) => {
+                switch (key) {
+                case 'smoke_alarm_state':
+                    await entity.read(
+                        'ssIasZone',
+                        [0xf0],
+                        manufacturerOptions,
+                    );
+                    break;
+                case 'intruder_alarm_state':
+                    await entity.read(
+                        'ssIasZone',
+                        [0xf0],
+                        manufacturerOptions,
+                    );
+                    break;
+                default: // Unknown key
+                    throw new Error(
+                        `Unhandled key toZigbee.rbshoszbeu.convertGet ${key}`,
+                    );
+                }
+            },
+        }],
         meta: {},
         configure: async (device, coordinatorEndpoint, logger) => {
             const endpoint = device.getEndpoint(1);
-            await reporting.bind(endpoint, coordinatorEndpoint, ['genPowerCfg', 64684]);
+            await reporting.bind(endpoint, coordinatorEndpoint, ['genPowerCfg',
+                'ssIasZone',
+                'ssIasWd',
+                'genBasic',
+                64684]);
             await reporting.batteryPercentageRemaining(endpoint);
+            device.defaultSendRequestWhen = 'immediate';
+            device.save();
+            await endpoint.unbind('genPollCtrl', coordinatorEndpoint);
         },
-        exposes: [e.smoke(), e.battery(), e.battery_low(), e.test()],
+        exposes: [e.smoke(), e.battery(), e.battery_low(), e.test(),
```

Comment:
Is it possible to turn on both smoke and intruder sound on at the same time? If not I think it makes more sense to have 1 expose. (an `e.enum()`)

Reply:
Yes it is possible. The sound gets queued.
Ex: turning on smoke and intruder. Turning off smoke results in the intruder alarm being played. They both require a different 'OFF' value to be send.

To turn of smoke, value **46080** needs to be send, and for intruder value: **46081** needs to be send.

I believe the enum falls short here. The convertSet could send out 2 commands but that feels hacky too.

What is your suggestion? @Koenkk

Reply:
Then we should keep it as is (2 separate exposes)

Reply:
Ok ready for merge I believe.